crates/cubecl-core/src/ir/scope.rs

```diff
@@ -96,77 +98,58 @@ impl Scope {
 
     /// Create a matrix variable
     pub fn create_matrix(&mut self, matrix: Matrix) -> Variable {
-        let index = self.matrices.len() as u16;
```

Comment:
This functions is never used. It is safe to remove it?

Reply:
Probably, I think it's just there for legacy support with `cpa!`. Don't think there's really a point in keeping it though, since cpa kernels are deprecated and I doubt we'll need this for anything like bounds checking.